### PR TITLE
feat: working PoC for map transitions

### LINKED ISSUE #24: feat: map transitions (being able to go from map to map)

### PR DESCRIPTION
**Pull request by zwindler**
Fixes #24

### CODE REVIEW

File: maps/maps_logic.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+	// If the object was found, remove it from the slice
+	if indexToRemove >= 0 {
+		currentMap.ObjectList = append(currentMap.ObjectList[:indexToRemove], currentMap.ObjectList[indexToRemove+1:]...)
+	}
+}
+
+// For a given map, remove NPC by list id and hide CanvasImage
+func (currentMap *Map) RemoveNPC(npcToRemove *model.NPCStats) {
+	var indexToRemove int = -1
```
Comment (azrod) [inline maps/maps_logic.go:108]: In the [excellent guide](https://github.com/uber-go/guide/blob/master/style.md#local-variable-declarations)provided by Uber it is not recommended to use `var xx = value` for set value. 

> Short variable declarations (:=) should be used if a variable is being set to some value explicitly.

```suggestion
	var indexToRemove int = -1
	indexToRemove := -1
```

File: maps/maps_logic.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	// TODO: add info about NPCs in maps for fixed maps
+	// for generated maps, I'll have to create this randomly
+
+	// Loop through the NPC data slice and create/draw each NPC
+	for _, data := range currentMap.spawnNPC {
+		npc := model.CreateNPC(data.npc, data.x, data.y)
+		currentMap.NPCList = append(currentMap.NPCList, npc)
+	}
+
+}
```
Comment (azrod) [inline maps/maps_logic.go:88]: Other suggest. If your give `NPCList` a Type 
```go
type mapNPCList []*model.NPCStats

type Map struct {
	Name           string
	PlayerStart    Coord
	spawnNPC       SpawnNPC
	NPCList        mapNPCList
	ObjectList     []*model.Object
	MapMatrix      [][]int
	MapTransitions []SpecialTile
}
```
It is possible to create new methods for the slice.
Some examples 

```go
// Append adds a new element *model.NPCStats to the end of the slice mapNPCList
func (list *mapNPCList) Append(npc *model.NPCStats) {
	*list = append(*list, npc)
}

// Unhook removes the element at index i from the slice mapNPCList
func (list *mapNPCList) Unhook(i int) {
	*list = append((*list)[:i], (*list)[i+1:]...)
}

// IsHidden returns true if the NPC is hidden
func (list *mapNPCList) IsHidden(index int) bool {
	return (*list)[index].Avatar.CanvasImage.Hidden
}

// SetHidden sets the NPC hidden status
func (list *mapNPCList) SetHidden(index int, hidden bool) {
	(*list)[index].Avatar.CanvasImage.Hidden = hidden
}
```

```suggestion
// AddNPCs adds NPCs on a map from spawnNPC struct
func (currentMap *Map) AddNPCs() {
	// TODO: add info about NPCs in maps for fixed maps
	// for generated maps, I'll have to create this randomly

	// Loop through the NPC data slice and create/draw each NPC
	for _, data := range currentMap.spawnNPC {
	        currentMap.NPCList.Append(model.CreateNPC(data.npc, data.x, data.y))
	}

}
```

```go
// For a given map, remove NPC by list id and hide CanvasImage
func (currentMap *Map) RemoveNPC(npcToRemove *model.NPCStats) {
	var indexToRemove int = -1
	for i, npc := range currentMap.NPCList {
		if npc == npcToRemove {
			indexToRemove = i
			break
		}
	}

	// If the npc was found, remove it from the slice
	if indexToRemove >= 0 {
		// remove NPC image from fyne map
		currentMap.NPCList.SetHidden(indexToRemove, true)
		// remove NPC from NPCList slice
		currentMap.NPCList.Unhook(indexToRemove)
	}
}
```

**Comment by zwindler** [comment]
WIP load/save is broken

Also, I have to save AllTheMaps rather than currentMap, & I need to add some info to determine on which map the player is

File: maps/maps_logic.go (hunk 2 above)
Comment (zwindler) [reply maps/maps_logic.go]: I remove SpawnNPC completely, it's a pain in the *** to have SpawnNPC + NPCList

**Comment by zwindler** [comment]
Still WIP but progressing

For some reason, NPCs and Objects don't have CanvasImage after "load game", while player does

**Comment by zwindler** [comment]
Working version :)

**Pull request merged**